### PR TITLE
Specify that image upload title and message are optional

### PR DESCRIPTION
**Pull request by DamascenoRafael**
Add “optional” in the title's placeholder.
And since the message is a TextView I added UITextViewDelegate methods to change text and textColor when needed.

ref #771

### CODE REVIEW

**Comment by Sherlouk** [comment]
Sorry, I may be missing something - Why are we not using the placeholder attribute that text fields already provide us with?

**Comment by DamascenoRafael** [comment]
Hey @Sherlouk, I'm using the placeholder attribute in the title (titleTextField), but the message (bodyTextField) is actually a UITextView and not a UITextField. I can change the name of the outlet if you want, I think it would be better if we called bodyTextView.

**Comment by Sherlouk** [comment]
Ahh nah I just didn't realise there was no placeholder on text view 😅 Assumed it would be there, my bad!

**Comment by DamascenoRafael** [comment]
That's ok 😉

File: Classes/Image Upload/ImageUploadTableViewController.swift [hunk 1]
```diff
@@ -186,13 +196,35 @@ class ImageUploadTableViewController: UITableViewController, UITextFieldDelegate
             }
         }
     }
+    
+}
 
-    // MARK: UITextFieldDelegate
+// MARK: UITextFieldDelegate
 
+extension ImageUploadTableViewController: UITextFieldDelegate {
```
Comment (rnystrom) [inline Classes/Image Upload/ImageUploadTableViewController.swift]: Can we actually avoid adding extensions w/in the same file? Trying to keep the code style consistent.

File: Classes/Image Upload/ImageUploadTableViewController.swift [hunk 2]
```diff
@@ -186,13 +196,35 @@ class ImageUploadTableViewController: UITableViewController, UITextFieldDelegate
...
         bodyTextField.becomeFirstResponder()
         return false
     }
+}
 
+// MARK: UITextViewDelegate
+
+extension ImageUploadTableViewController: UITextViewDelegate {
+    func textViewDidBeginEditing(_ textView: UITextView) {
+        if textView.text == bodyPlaceholder {
```
Comment (rnystrom) [inline Classes/Image Upload/ImageUploadTableViewController.swift]: Maybe we should just start w/ `Optional` in the text field and add this later if totally necessary. Doing this text comparison is pretty inefficient. `SLKTextView` has an implementation for placeholders w/ `UITextView` already too.

File: Classes/Image Upload/ImageUploadTableViewController.swift (hunk 1 above)
Comment (BasThomas) [reply Classes/Image Upload/ImageUploadTableViewController.swift]: I did the same in #746 though, fwiw

File: Classes/Image Upload/ImageUploadTableViewController.swift (hunk 2 above)
Comment (DamascenoRafael) [reply Classes/Image Upload/ImageUploadTableViewController.swift]: I didn't know about `SLKTextView`. I'll take a look and try to do with it, ok?

Comment (BasThomas) [reply Classes/Image Upload/ImageUploadTableViewController.swift]: See here: https://github.com/slackhq/SlackTextViewController/blob/master/Source/SLKTextView.h

File: Classes/Image Upload/ImageUploadTableViewController.swift (hunk 1 above)
Comment (DamascenoRafael) [reply Classes/Image Upload/ImageUploadTableViewController.swift]: @rnystrom Sure! But I've seen some style guides recommending this way - https://github.com/raywenderlich/swift-style-guide#protocol-conformance

Comment (DamascenoRafael) [reply Classes/Image Upload/ImageUploadTableViewController.swift]: I changed the SLKTextView. There's still the UITextFieldDelegate extension, do you want me to put it without the extension?

Comment (rnystrom) [reply Classes/Image Upload/ImageUploadTableViewController.swift]: I actually ran into compilation issues b/c of overuse of `extension` and ObjC interop, so I'm just moving away from them as much as possible 😋 .

**Pull request merged**